coderd/workspaceapps/db.go

```diff
@@ -5,29 +5,76 @@ import (
 	"database/sql"
 	"fmt"
 	"net/http"
-	"strings"
+	"net/url"
 	"time"
 
 	"golang.org/x/xerrors"
 
 	"cdr.dev/slog"
+
 	"github.com/coder/coder/coderd/database"
 	"github.com/coder/coder/coderd/database/dbauthz"
 	"github.com/coder/coder/coderd/httpapi"
 	"github.com/coder/coder/coderd/httpmw"
 	"github.com/coder/coder/coderd/rbac"
 	"github.com/coder/coder/codersdk"
-	"github.com/coder/coder/site"
 )
 
-const (
-	// TODO(@deansheather): configurable expiry
-	TicketExpiry = time.Minute
+// DBTicketProvider provides authentication and authorization for workspace apps
+// by querying the database if the request is missing a valid ticket.
+type DBTicketProvider struct {
+	Logger slog.Logger
+
+	AccessURL                     *url.URL
```

Comment:
Can we comment what this access url is for? It is just for displaying an error page that has a link to the dashboard. So just mention this access url should be the url to the coderd dashboard.

For external workspace proxies, they will not use their own access url, but the primary url.